libnetwork/drivers/bridge/bridge_linux.go

```diff
@@ -735,6 +735,14 @@ func (d *driver) createNetwork(config *networkConfiguration) (err error) {
 		bridgeSetup.queueStep(setupDefaultSysctl)
 	}
 
+	// Always set the bridge's MTU if specified. This is purely cosmetic; a bridge's
+	// MTU is the min MTU of device connected to it, and MTU will be set on each
+	// 'veth'. But, for a non-default MTU, the bridge's MTU will look wrong until a
+	// container is attached.
```

Comment:
This is not purely cosmetic: on Linux 4.17 and newer, manually setting the MTU disables MTU auto tuning. https://github.com/torvalds/linux/commit/804b854d374e39f5f8bff9638fd274b9a9ca7d33

Reply:
The MTU is defined for the network, and all the endpoints attached to the bridge get that MTU. So, auto tuning would select the network's MTU ... and, I thought, the only effect of setting it explicitly on the bridge would be to have it report that MTU before an endpoint was attached (so, cosmetic in that sense).

I don't have laptop access at the moment  (won't be properly back online until Tues) - but if this change is shown to be causing problems, we should just revert it.

Reply:
Sorry, I'm speaking in a more general sense that setting the MTU on a Linux Ethernet Bridge is sticky on newer kernels, whereas on older kernels it would get clobbered every time a link is attached to it. Since we control all the links and set the MTU on all of them to the bridge MTU, it's a distinction without a difference _to the libnetwork bridge driver_ unless someone manually changes the MTU of host-side veths. I don't think we need to revert [once we patch the bridge driver `setupMTU()` to ignore `-EINVAL`](https://github.com/moby/moby/issues/47308#issuecomment-1924193313).

Reply:
Forgive the potential dumb question as I am not well versed in linux networking internals.

What will be end result behavior on older version of linux after patching that function to ignore the `EINVAL` error response ? Will it be like it was before this MR and the `docker0` interface will switch to the desired MTU > 1500 once a container is attached ?

Reply:
> Will it be like it was before this MR and the `docker0` interface will switch to the desired MTU > 1500 once a container is attached ?

Yes, exactly